### PR TITLE
Simplify CI detection

### PR DESCRIPTION
**Pull request by jakebailey**
Pulled out of #50597.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
     - run:  npm ci
 
     - name: Linter
-      run:  npm run lint:ci
+      run:  npm run
```
Comment (andrewbranch) [inline .github/workflows/ci.yml:49]: ?

Comment (jakebailey) [reply .github/workflows/ci.yml:49]: Oh oops, I forgot to grab the second commit off the other PR.

**Review by andrewbranch — APPROVED** [review]

File: scripts/build/options.js [hunk 2]
```diff
@@ -2,9 +2,11 @@
 const minimist = require("minimist");
 const os = require("os");
 
+const ci = ["1", "true"].includes(process.env.CI);
```
Comment (amcasey) [inline scripts/build/options.js:5]: This gets set by the workflow or something, right?  It's not something _we_ set elsewhere?

**Review by amcasey — APPROVED** [review]
Seems like this will make it more painful to run the CI lint rules locally but it also seems like no one cares.

**Comment by jakebailey** [comment]
> Seems like this will make it more painful to run the CI lint rules locally but it also seems like no one cares.

What do you mean? We don't have any rules that are CI specific anymore, so `npm run lint` is equivalent to `npm run lint -- --ci` or what was `npm run lint:ci`. The only thing `--ci` changes is the reporter used.

File: scripts/build/options.js (hunk 2 above)
Comment (jakebailey) [reply scripts/build/options.js:5]: Correct; GHA sets `CI=1` or similar.

File: Gulpfile.js [hunk 3]
```diff
@@ -372,9 +372,6 @@ const lint = eslint(".");
 lint.displayName = "lint";
 task("lint", lint);
 task("lint").description = "Runs eslint on the compiler and scripts sources.";
-task("lint").flags = {
-    "   --ci": "Runs eslint additional rules",
```
Comment (jakebailey) [inline Gulpfile.js:376]: For posterity, there are no additional rules; all rules already run. Probably this is a leftover from tslint or something.

**Pull request merged**